pyqir-evaluator/tests/test_unknown_external_func.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from pyqir.evaluator import GateLogger, NonadaptiveEvaluator
+import pytest
+import tempfile
+
+
+def test_unknown_external_func():
+    content = """
+        ; ModuleID = 'test_unknown_external_func'
+        source_filename = "test_unknown_external_func"
+
+        %String = type opaque
+
+        declare %String* @__quantum__rt__bool_to_string(i1)
+
+        define void @main() #1 {
+        entry:
+            call %String* @__quantum__rt__bool_to_string(i1 1)
+            ret void
+        }
+
+        attributes #1 = { "EntryPoint" }
+    """
+
+    evaluator = NonadaptiveEvaluator()
+    logger = GateLogger()
+    with pytest.raises(OSError):
```

Comment:
Is OSError the correct type for the error?

Reply:
It's what we're currently raising. https://github.com/qir-alliance/pyqir/blob/main/pyqir-evaluator/src/python.rs#L101
It's probably not the best choice for every possible error path (e.g multiple entry points in module) but changing this would require keeping information from where the error originated (and not just a string) so we can raise the appropriate error type in Python. I feel it's a bit outside of the scope of this PR.